src/main/java/com/medbay/service/TherapyService.java

```diff
@@ -50,11 +50,58 @@ public ResponseEntity<Void> createTherapy(CreateTherapyRequest request){
         therapyRepository.save(therapy);
         return ResponseEntity.ok().build();
     }
+
+ */
     public ResponseEntity<Void> deleteTherapy(Long id){
         if(!therapyRepository.existsById(id)) {
             return ResponseEntity.notFound().build();
         }
         therapyRepository.deleteById(id);
         return ResponseEntity.ok().build();
     }
+
+    public ResponseEntity<Void> createNewTherapy(CreateTherapyRequest request) {
+      //  Optional<Patient> patientOptional = patientRepository.findById(request.getPatientId());
+        //Optional<Employee> employeeOptional = employeeRepository.findById(request.getEmployeeId());
+
+        if (request.getTherapyCode().isEmpty() || request.getAppointmentDate().toString().isEmpty()) {
+            return ResponseEntity.status(HttpStatus.BAD_REQUEST).build();
+        }
+
+        LocalDateTime appointmentDateTime = request.getAppointmentDate();
+TherapyType therapyType = therapyTypeRepository.findByTherapyCode(request.getTherapyCode());
+        Therapy therapy = Therapy.builder()
+                .therapyStatus(TherapyStatus.PENDING)
+                .therapyType(therapyType)
+                .build();
+
+     User user = (User) SecurityContextHolder.getContext().getAuthentication().getPrincipal();
+        Patient patient = (Patient)user;
+        // Dodajem terapiju pacijentu
+        patient.addTherapy(therapy);
+
+        //Dodajem terapiju employee, na lraju ne znam jel se nama salje employeeId pa ne stvaram to tu
+        //employee.addTherapy(therapy);
+
+        // Spremi ažuriranog pacijenta u repozitorij
+        patientRepository.save(patient);
+
+        // Slanje zahtjeva adminu
+        sendRequestToAdmin(request.getTherapyCode(), appointmentDateTime, patient.getMBO());
+
+        return ResponseEntity.ok().build();
+    }
+
+    private void sendRequestToAdmin(String therapyCode, LocalDateTime therapyDate, String patientMBO) {
+        boolean verify = false;
+        //tu tzreba provjera zeli li admin odobriti, ne znam kako su to na frontu planirali simulirati pa sam samo postavila boolean verify
+        TherapyType therapyType = therapyTypeRepository.findByTherapyCode(therapyCode);
+        Therapy therapy = therapyRepository.findByTherapyType(therapyType);
+
```

Comment:
Tebi ce front poslati VERIFIED ili REJECTED i trebali bi poslat razlog zasto, ti bis to trebala zapisati u bazi i onda poslati mail, zasad nek bude isti i kad je odibjen ili kad je primljen